### DISCUSSION TITLE
Style of `#include` statements

### DISCUSSION THREAD

**Discussion by yschroeder97** [nebulastream/nebulastream#542] [General]
To get a quick overview of what headers are included (and therefore what symbols from other files are used) we could follow a more structured approach like the one from [google](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes).

They structure includes the following way for a `file.cpp`:
```c++
#include <related_header.hpp>

#include <fcntl.h>

#include <vector>

#include "fmt/format.h"

#include "other_project_file.hpp"
```

- Grouping related includes separated by blank lines
1. Related header
2. C and OS headers
3. C++ standard library headers
4. Project headers
- Prefer quotes "" to angular brackets <> wherever possible (OS headers, C headers, and standard library headers)
- Sort alphabetically within each group

We could incorporate the style into `.clang-format` and apply the style automatically.

**Comment by keyseven123** [comment]
For the order, I do not care. Question is what is the difference between related and project headers?


> Prefer quotes "" to angular brackets <> wherever possible (OS headers, C headers, and standard library headers)

Before we do this, we should have a discussion on how to make sure that not everything gets included from anywhere in our system. One reason we went with `<>` instead of "" is that it is a quick way of checking if someone includes something from a different module. I just fear that if we rush the decision of `""`, we will have a completely mess in 2-3 months as certain files are simply being included anywhere in the system.
